core/src/main/java/com/microsoft/applicationinsights/internal/processor/RequestTelemetryFilter.java

```diff
@@ -0,0 +1,150 @@
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.internal.processor;
+
+import java.util.Set;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import com.microsoft.applicationinsights.agent.internal.common.StringUtils;
+import com.microsoft.applicationinsights.extensibility.TelemetryProcessor;
+import com.microsoft.applicationinsights.internal.annotation.BuiltInProcessor;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.telemetry.Duration;
+import com.microsoft.applicationinsights.telemetry.RequestTelemetry;
+import com.microsoft.applicationinsights.telemetry.Telemetry;
+
+/**
+ * The class can filter out RequestTelemetries that
+ *   have a duration which is less than a predefined value
+ *   have http codes that are not needed based on configuration
+ *
+ *  Illegal value will prevent from the filter from being used.
+ *
+ * Created by gupele on 7/26/2016.
+ */
+@BuiltInProcessor("RequestTelemetryFilter")
+public final class RequestTelemetryFilter implements TelemetryProcessor {
+    private final class FromTo {
+        public final int from;
+        public final int to;
+
+        private FromTo(int from, int to) {
+            this.from = from;
+            this.to = to;
+        }
+    }
+
+    private long minimumDurationInMS = 0;
+    private boolean hasBlocked;
+    private final Set<String> exactBadResponseCodes = new HashSet<String>();
+    private final List<FromTo> fromTos = new ArrayList<FromTo>();
+
+    public RequestTelemetryFilter() {
+    }
+
+    @Override
+    public boolean process(Telemetry telemetry) {
+        if (telemetry == null) {
+            return true;
+        }
+
+        if (!hasBlocked && minimumDurationInMS <= 0) {
+            return true;
+        }
+
+        if (telemetry instanceof RequestTelemetry) {
+            RequestTelemetry requestTelemetry = (RequestTelemetry)telemetry;
+            String requestRespohseCode = requestTelemetry.getResponseCode();
```

Comment:
maybe responseCode instead fo requestResponseCode? The combination of request and response together is a bit confusing (at least to me :-) )